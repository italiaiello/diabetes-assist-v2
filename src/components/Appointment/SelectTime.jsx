import React, { useState } from 'react'
import TimeSlot from './TimeSlot'
import HealthProfessionals from '../../json/test.json'

const SelectTime = props => {

    const professionalsArray = Object.values(HealthProfessionals[props.professionalId - 1].availableTimes)
    const timeSlots = [10, 11, 12, 1, 2, 3, 4]
    const [timeSlotId, setTimeSlotId] = useState(-1)


    return (
        <div id="timeSelect">
            {
                professionalsArray.map((available, i) => {
                    return (
                        available === "Yes" ?
                        <TimeSlot key={i} availability={available} 
                                        time={timeSlots[i]} 
                                        setTimeSlot={props.setTimeSlot}
                                        id={i}
                                        timeSlotId={timeSlotId}
                                        setTimeSlotId={setTimeSlotId}
                        />
                        :
                        <TimeSlot key={i} availability={available} 
                                        time={timeSlots[i]}
                                        id={i}
                        />
                    )
                })
            }
        </div>
    )
}

export default SelectTime