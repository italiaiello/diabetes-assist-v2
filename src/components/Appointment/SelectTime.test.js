import React from 'react';
import { shallow } from 'enzyme';
import SelectTime from './SelectTime';

describe('SelectTime.js', () => {

  let wrapper;
  
  const mockProfessionalId = 1

  beforeEach(() => {
    wrapper = shallow(<SelectTime professionalId={mockProfessionalId} />);
  });

  it('renders', () => {
    expect(wrapper).toMatchSnapshot()
  })

})