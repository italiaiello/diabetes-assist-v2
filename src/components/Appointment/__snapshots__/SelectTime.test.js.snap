// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectTime.js renders 1`] = `ShallowWrapper {}`;
